### PR TITLE
Add Custom Text component

### PR DESCRIPTION
**Pull request by IgnacioAbadie**
### Summary
- Added Custom Text Component

###Example View

screen shot 2017-08-08 at 2 48 38 pm

Example Code
screen shot 2017-08-08 at 2 48 10 pm

### CODE REVIEW

File: generators/app/templates/src/app/components/CustomText/styles.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import { StyleSheet } from 'react-native';
+
+import { blue, white, green, gray, black, transparent } from '../../../utils/colors';
+import { IS_SMALL_DEVICE } from '../../../utils/constants';
+
+const amountToResize = 2 * IS_SMALL_DEVICE;
+
+export default StyleSheet.create({
+  base: {
+    fontFamily: 'poppins',
```
Comment (ignaciosantise) [inline generators/app/templates/src/app/components/CustomText/styles.js]: This font is not available in this project! Remove this line to solve the red screen of death 🔴

Comment (IgnacioAbadie) [inline generators/app/templates/src/app/components/CustomText/styles.js]: Mery Poppins style 😞

File: generators/app/templates/src/app/components/CustomText/index.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  }
+}
+
+CustomText.defaultProps = {
+  textProps: {}
+};
+
+CustomText.propTypes = {
+  children: PropTypes.node.isRequired,
+  textProps: PropTypes.shape({ ...Text.propTypes })
```
Comment (mvbattan) [inline generators/app/templates/src/app/components/CustomText/index.js]: `textProps: Text.propTypes` ?

File: generators/app/templates/src/app/components/CustomText/styles.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+const amountToResize = 2 * IS_SMALL_DEVICE;
+
+// Custom Text Style
+// You can add styles to Base like Family Font to be the Text styles base!
+// If you wanna a particular style you can add this like colors or sizes and then you have to
+// add the prop into index.js
+
+export default StyleSheet.create({
+  base: {
+    fontSize: 16 - amountToResize,
```
Comment (mvbattan) [inline generators/app/templates/src/app/components/CustomText/styles.js]: Extract this `16` to a constants file. I'd place this in a `fonts` folder.

File: generators/app/templates/src/app/components/CustomText/styles.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    fontSize: 13 - amountToResize
+  },
+  medium: {
+    fontSize: 14 - amountToResize
+  },
+  big: {
+    fontSize: 20 - amountToResize
+  },
+  xbig: {
+    fontSize: 37 - amountToResize
```
Comment (mvbattan) [inline generators/app/templates/src/app/components/CustomText/styles.js]: These numbers also should belong to a constant file

File: generators/app/templates/src/app/components/CustomText/styles.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
+import { StyleSheet } from 'react-native';
+
+import { blue, white, green, gray, black, transparent } from '../../../utils/colors';
+import { IS_SMALL_DEVICE } from '../../../utils/constants';
+
+const amountToResize = 2 * IS_SMALL_DEVICE;
+
+// Custom Text Style
+// You can add styles to Base like Family Font to be the Text styles base!
+// If you wanna a particular style you can add this like colors or sizes and then you have to
```
Comment (ignaciosantise) [inline generators/app/templates/src/app/components/CustomText/styles.js]: `if you want to add a custom style, you need to add it here and in VARIANTS Array in index.js` ?

File: generators/app/templates/src/app/components/CustomText/styles.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
+import { StyleSheet } from 'react-native';
+
+import { blue, white, green, gray, black, transparent } from '../../../utils/colors';
+import { IS_SMALL_DEVICE } from '../../../utils/constants';
```
Comment (mvbattan) [inline generators/app/templates/src/app/components/CustomText/styles.js]: Mmm, `utils/constants` can become a `god file`. I'll raise an issue to address this refactor.

File: generators/app/templates/src/utils/constants.js [hunk 7]
```diff
@@ -15,3 +15,5 @@ export const WINDOW_WIDTH = windowDimensions.width;
 
 export const VIEWPORT_HEIGHT =
   WINDOW_HEIGHT - TABBAR_HEIGHT - ACTION_BAR_HEIGHT - (STATUS_BAR_IS_FIXED ? STATUS_BAR_HEIGHT : 0);
+
+export const IS_SMALL_DEVICE = WINDOW_HEIGHT < 570;
```
Comment (mvbattan) [inline generators/app/templates/src/utils/constants.js]: 🤔  landscape rotation might break this if supported

File: generators/app/templates/src/app/components/CustomText/styles.js (hunk 4 above)
Comment (mvbattan) [inline generators/app/templates/src/app/components/CustomText/styles.js]: Is there a way to reduce these duplicated declarations ? Two patterns appears here:
```
import { colorName } from '../etc/colors';
colorName: {
  color: colorName
}

import { fontSizePx } from '../etc/fonts/sizes'

fontSizePx: {
  fontSize: fontSizePx - amountToResize
}
```

File: generators/app/templates/src/app/components/CustomText/index.js (hunk 2 above)
Comment (sbalay) [inline generators/app/templates/src/app/components/CustomText/index.js]: A value assigned as a propTypes cannot be a plain js object, it needs to be a propTypes instance.
That's why `textProps: Text.propTypes` will not work, and `textProps: PropTypes.shape({ ...Text.propTypes })` is needed

File: generators/app/templates/src/app/components/CustomText/styles.js (hunk 6 above)
Comment (sbalay) [inline generators/app/templates/src/app/components/CustomText/styles.js]: yep, we can work on a refactor in a different pull request

File: generators/app/templates/src/app/components/CustomText/styles.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+const amountToResize = 2 * IS_SMALL_DEVICE;
+
+// Custom Text Style
```
Comment (sbalay) [inline generators/app/templates/src/app/components/CustomText/styles.js]: I'd write this as comments of the component, here may be more difficult to find.
```js
/**
 * You can add styles to Base like Family Font to be the Text styles base!
......
 */
export default class CustomText extends PureComponent {
```

File: generators/app/templates/src/app/components/CustomText/styles.js (hunk 4 above)
Comment (sbalay) [inline generators/app/templates/src/app/components/CustomText/styles.js]: If we think of an abstaction here we should apply the same in the VARIANTS declaration

Comment (sbalay) [inline generators/app/templates/src/app/components/CustomText/styles.js]: I see your point but I'd still keep it this way. These base styles will probably be changed after the project kickoff, for example: you may want that all the `xbig` texts have a `bold` font. Or the `red` texts to be `underlined` for errors.
I prefer declaring this styles in a very explicit way, so they are easier to refactor later.

File: generators/app/templates/src/utils/constants.js (hunk 7 above)
Comment (sbalay) [inline generators/app/templates/src/utils/constants.js]: Good catch. I'm not very convinced by this approach either

File: generators/app/templates/src/app/components/CustomText/index.js (hunk 2 above)
Comment (IgnacioAbadie) [reply generators/app/templates/src/app/components/CustomText/index.js]: :+1:

File: generators/app/templates/src/utils/constants.js (hunk 7 above)
Comment (IgnacioAbadie) [reply generators/app/templates/src/utils/constants.js]: @sbalay @mvbattan 
I would keep it that way until we found a better way to handle the sizes
maybe can add a comment

File: generators/app/templates/src/app/components/CustomText/styles.js (hunk 8 above)
Comment (IgnacioAbadie) [reply generators/app/templates/src/app/components/CustomText/styles.js]: :+1:

**Comment by IgnacioAbadie** [comment]
@mvbattan can i merge this PR?

**Pull request merged**